features/FEATURE_BLE/ble/generic/SecurityDb.h

```diff
@@ -584,28 +584,32 @@ class SecurityDb {
         WhitelistDbCb_t cb,
         ::Gap::Whitelist_t *whitelist
     ) {
-        for (size_t i = 0; i < get_entry_count() && i < whitelist->capacity; i++) {
+        for (size_t i = 0; i < get_entry_count() && whitelist->size < whitelist->capacity; i++) {
```

Comment:
Should `whitelist->size` be initialised to 0? Or is it the user's responsibility?

Reply:
whitelist is user provided, could already have something in it, would be 0 on a newly created one